pkg/database/database.go

```diff
@@ -99,3 +109,36 @@ func newJSONHandle(cipher encryption.Cipher) *codec.JsonHandle {
 	h.SetInterfaceExt(reflect.TypeOf((*api.SecureString)(nil)), 1, secureStringExt{cipher: cipher})
 	return h
 }
+
+func cosmosDB(ctx context.Context, im instancemetadata.InstanceMetadata) (string, string, error) {
+	rpAuthorizer, err := env.RPAuthorizer(azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		return "", "", err
+	}
+
+	databaseaccounts := documentdb.NewDatabaseAccountsClient(im.SubscriptionID(), rpAuthorizer)
+
+	accts, err := databaseaccounts.ListByResourceGroup(ctx, im.ResourceGroup())
+	if err != nil {
+		return "", "", err
+	}
+
+	if len(*accts.Value) != 1 {
+		return "", "", fmt.Errorf("found %d database accounts, expected 1", len(*accts.Value))
+	}
+
+	keys, err := databaseaccounts.ListKeys(ctx, im.ResourceGroup(), *(*accts.Value)[0].Name)
+	if err != nil {
+		return "", "", err
+	}
+
+	return *(*accts.Value)[0].Name, *keys.PrimaryMasterKey, nil
+}
+
+func databaseName(_env env.Lite) string {
+	if _env.Type() == env.Dev {
+		return os.Getenv("DATABASE_NAME")
```

Comment:
im not sure how I feel about `os.Getenv` scattered around the code base. If we do this we either need good documentation of each variable or better to try something like `envconfig` where we can validate if in Devmode `DATABASE_NAME` is set. Otherwise we will get random errors which will come from random places (in this case database client) which is hard to debug if you dont know code base very well.

What I'm trying to say, that we are dropping single place of configuration for scattered config. I dont feel very good about this :/

Reply:
I agree.  This is still in exploration phase and what you raise is definitely one of the problems with this PR.